src/classes/ExpectSObject.cls

```diff
@@ -0,0 +1,84 @@
+public class ExpectSObject extends Expect {
+    private static Boolean isSuccess = true;
+    private static Boolean isFail = false;
+    private static String exceptionString;
+    private final SObject expected;
+
+    public ExpectSObject(SObject expected) {
+        this.expected = expected;
+    }
+
+    protected override String getExpectedAsString() {
+        return String.valueOf(expected);
+    }
+
+    public ExpectSObject andIt { get { return this; } }
+
+    public ExpectSObject shouldEqual(SObject actual) {
+        shouldEqual(actual, expectedMessage(actual,''));
+        return this;
+    }
+
+    public ExpectSObject shouldEqual(SObject actual, String message) {
+        assert(expected === actual, message);
+        return this;
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual) {
+        return shouldNotEqual(actual, expectedMessage(actual,' not'));
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual, String message) {
+        assert(expected !== actual, message);
+        return this;
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObject actual) {
+        return sObjectTypeEquals(actual, expectedMessage(actual.getSObjectType(),''));
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObject actual, String message) {
+        assert(expected.getSObjectType() == actual.getSObjectType(), message);
+        return this;
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObjectType objType) {
+        return sObjectTypeEquals(objType, expectedMessage(objType,''));
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObjectType objType, String message) {
+        assert(expected.getSObjectType() == objType, message);
+        return this;
+    }
+
+    public ExpectSObject whenTryingToUpsert() {
```

Comment:
how do you feel about rephrasing as shouldFailToUpsert() shouldFailToUpsertWithMessage(message) ?